ledger/src/bank_forks.rs

```diff
@@ -177,34 +177,54 @@ impl BankForks {
             .last()
             .map(|bank| bank.transaction_count())
             .unwrap_or(0);
-
-        root_bank.squash();
-        let new_tx_count = root_bank.transaction_count();
-
-        // Generate a snapshot if snapshots are configured and it's been an appropriate number
-        // of banks since the last snapshot
+        // Generate each snapshot at a fixed interval
+        let mut squashed_slot = None;
         if self.snapshot_config.is_some() && snapshot_package_sender.is_some() {
             let config = self.snapshot_config.as_ref().unwrap();
-            info!("setting snapshot root: {}", root);
-            if root - self.last_snapshot_slot >= config.snapshot_interval_slots as Slot {
-                let mut snapshot_time = Measure::start("total-snapshot-ms");
-                let r = self.generate_snapshot(
-                    root,
-                    &root_bank.src.roots(),
-                    snapshot_package_sender.as_ref().unwrap(),
-                );
-                if r.is_err() {
-                    warn!("Error generating snapshot for bank: {}, err: {:?}", root, r);
-                } else {
-                    self.last_snapshot_slot = root;
+            let mut banks = vec![root_bank];
+            let parents = root_bank.parents();
+            banks.extend(parents.iter());
+            for bank in banks.iter() {
+                let root = bank.slot();
+                if bank.block_height() % (config.snapshot_interval_slots as u64) == 0 {
```

Comment:
nits: how about rename `config.snapshot_interval_slots` to `config.snapshot_interval_block_heights`?

Reply:
Well it's still in slots, snapshot_interval_block_heights is a bit confusing to me especially if propagated to the command-line option. Let's think about, maybe there's a better name, we can rename in a follow-up.